### ISSUE TITLE
No tools:unzip_collection, Flatten collection and Sort

### ISSUE THREAD

**Issue by Dong2843997826** [qiime2/q2galaxy#52]
Hi，@ebolyen I used docker to pull and run q2galaxy. When building my workflow, I found that there are no basic tools, such as unzip_collection, Flatten collection and Sort. How can I install these tools in q2galaxy?

**Comment by Dong2843997826** [comment]
@Oddant1

**Comment by Oddant1** [comment]
Hello @Dong2843997826.

What use case do you have for unzipping collections? You should be able to use individual elements of collections as inputs to other actions as necessary.

Additionally, QIIME 2 does not currently nest collections, so flattening isn't needed functionality.

Sorting is a potential nice to have in the future, but order also does not currently matter.

**Comment by Dong2843997826** [comment]
@Oddant1.Thank you for your reply.Oh, I did add some other workflows.But I found that none of these tools, like unzip_collection, are available. If I want to use these, how should I add them?

**Comment by Oddant1** [comment]
@Dong2843997826 are you saying the built-in galaxy tools for unzipping, flattening, and sorting collections aren't there? [ex](https://usegalaxy.org/?tool_id=__FLATTEN__&version=latest)

If so, that would be an issue with your galaxy configuration itself that I'm not sure how to resolve. We don't have any tools that implement that functionality in QIIME 2.

**Comment by Dong2843997826** [comment]
Well, @Oddant1，@ebolyen，I have these tools when I use docker-galaxy-stable, but q2galaxy doesn't. I wonder if q2galaxy doesn't install these tools by default.

**Comment by ebolyen** [comment]
That makes sense, I think one issue is there hasn't been an update of the galaxy docker image. Ideally we would set up our image a bit differently so the default tools would come along (and we wouldn't have to overwrite them as you see).

I think it makes sense to wait on this a bit, as the entire docker stack is in pretty bad shape for this and we should be able to fix it really easily at some point(tm). 

@Dong2843997826, I would recommend using a public galaxy server instead of the docker image at this point. We need a more modern version of Galaxy to support QIIME 2 anyhow (the current image is a kind of hacked together proof of concept).
